docling_core/types/experimental/document.py

```diff
@@ -0,0 +1,1106 @@
+            "title",
+            "subtitle-level-1",
+            "paragraph",
+            "caption",
+            "table",
+            "Text",
+            "text",
+        ],
+        strict_text: bool = False,
+    ) -> str:
+        r"""Serialize to Markdown.
+
+        Operates on a slice of the document's main_text as defined through arguments
+        main_text_start and main_text_stop; defaulting to the whole main_text.
+
+        :param delim: Delimiter to use when concatenating the various
+                Markdown parts. Defaults to "\n\n".
+        :type delim: str
+        :param from_element: Body slicing start index (inclusive).
+                Defaults to 0.
+        :type from_element: int
+        :param to_element: Body slicing stop index
+                (exclusive). Defaults to None.
+        :type to_element: Optional[int]
+        :param delim: str:  (Default value = "\n\n")
+        :param from_element: int:  (Default value = 0)
+        :param to_element: Optional[int]:  (Default value = None)
+        :param labels: list[str]:  (Default value = ["title")
+        :param "subtitle-level-1":
+        :param "paragraph":
+        :param "caption":
+        :param "table":
+        :param "Text":
+        :param "text":
+        :param ]:
+        :param strict_text: bool:  (Default value = False)
+        :returns: The exported Markdown representation.
+        :rtype: str
+        """
+        has_title = False
+        prev_text = ""
+        md_texts: list[str] = []
+
+        skip_count = 0
+        for ix, (item, level) in enumerate(self.iterate_elements(self.body)):
+            if skip_count < from_element:
+                skip_count += 1
+                continue  # skip as many items as you want
+
+            if to_element and ix >= to_element:
+                break
+
+            markdown_text = ""
+
+            if isinstance(item, DocItem):
+                item_type = item.label
+
+                if isinstance(item, TextItem) and item_type in labels:
+                    text = item.text
+
+                    # ignore repeated text
+                    if prev_text == text or text is None:
+                        continue
+                    else:
+                        prev_text = text
+
+                    # first title match
+                    if item_type == "title" and not has_title:
+                        if strict_text:
+                            markdown_text = f"{text}"
+                        else:
+                            markdown_text = f"# {text}"
+                        has_title = True
+
+                    # secondary titles
+                    elif item_type in {"title", "subtitle-level-1"} or (
+                        has_title and item_type == "title"
+                    ):
+                        if strict_text:
+                            markdown_text = f"{text}"
+                        else:
+                            markdown_text = f"## {text}"
```

Comment:
Was forced to roll this exact stuff back in other places because the line length exceeded 88, and one _can not break_ these inline conditionals across lines.

Reply:
I'm fine to leave it like this and consider it as resolved. But I'd like to point out that:
- the inline conditional assignment alternative fits within the 88 limit
- the assignments can be broken with a `()` after `=` to gain a couple of characters